stage1_fly/run/main.go

```diff
@@ -0,0 +1,381 @@
+	p, err := stage1commontypes.LoadPod(root, uuid)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to load pod: %v\n", err)
+		return 1
+	}
+
+	if len(p.Manifest.Apps) != 1 {
+		log.Fatalf("Flavor %q only supports 1 application per Pod for now.", flavor)
+	}
+
+	// TODO: insert environment from manifest
+	env := []string{"PATH=/bin:/sbin:/usr/bin:/usr/local/bin"}
+	args := p.Manifest.Apps[0].App.Exec
+	rfs := filepath.Join(common.AppPath(p.Root, p.Manifest.Apps[0].Name), "rootfs")
+
+	// set close-on-exec flag on RKT_LOCK_FD so it gets correctly closed when invoking
+	// network plugins
+	lfd, err := common.GetRktLockFD()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to get rkt lock fd: %v\n", err)
+		return 1
+	}
+
+	if err := sys.CloseOnExec(lfd, true); err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to set FD_CLOEXEC on rkt lock: %v\n", err)
+		return 1
+	}
+
+	argFlyMounts, err := evaluateMounts(rfs, string(p.Manifest.Apps[0].Name), p)
+	if err != nil {
+		log.Fatalf("Error evaluating mounts: %v", err)
+	}
+
+	effectiveMounts := append(
+		[]flyMount{
+			{"", "", "/dev", "none", syscall.MS_REC | syscall.MS_SHARED},
+			{"/dev", rfs, "/dev", "none", syscall.MS_BIND | syscall.MS_REC},
+
+			{"", "", "/proc", "none", syscall.MS_REC | syscall.MS_SHARED},
+			{"/proc", rfs, "/proc", "none", syscall.MS_BIND | syscall.MS_REC},
+
+			{"", "", "/sys", "none", syscall.MS_REC | syscall.MS_SHARED},
+			{"/sys", rfs, "/sys", "none", syscall.MS_BIND | syscall.MS_REC},
+
+			{"tmpfs", rfs, "/tmp", "tmpfs", 0},
+		},
+		argFlyMounts...,
+	)
+
+	for _, mount := range effectiveMounts {
+		var (
+			err            error
+			hostPathInfo   os.FileInfo
+			targetPathInfo os.FileInfo
+		)
+		if mount.HostPath != "" && strings.HasPrefix(mount.HostPath, "/") {
+			if hostPathInfo, err = os.Stat(mount.HostPath); err != nil {
+				log.Fatalf("fly: something is wrong with the host directory %s: \n%v", mount.HostPath, err)
+			}
+		} else {
+			hostPathInfo = nil
+		}
+
+		absTargetPath := filepath.Join(mount.TargetPrefixPath, mount.RelTargetPath)
+		if absTargetPath != "/" {
+			if targetPathInfo, err = os.Stat(absTargetPath); err != nil && !os.IsNotExist(err) {
+				log.Fatalf("fly: something is wrong with the target directory %s: \n%v", absTargetPath, err)
+			}
+
+			switch {
+			case targetPathInfo == nil:
+				absTargetPathParent, _ := filepath.Split(absTargetPath)
+				if err := os.MkdirAll(absTargetPathParent, 0700); err != nil {
+					log.Fatalf("fly: could not create directory %q: \n%v", absTargetPath, err)
+				}
+				switch {
+				case hostPathInfo == nil || hostPathInfo.IsDir():
+					if err := os.Mkdir(absTargetPath, 0700); err != nil {
+						log.Fatalf("fly: could not create directory %q: \n%v", absTargetPath, err)
+					}
+				case !hostPathInfo.IsDir():
+					file, err := os.OpenFile(absTargetPath, os.O_CREATE, 0700)
+					if err != nil {
+						log.Fatalf("fly: could not create file %q: \n%v", absTargetPath, err)
+					}
+					file.Close()
```

Comment:
I'm not sure about ignoring a close error here. If it is on purpose, consider using `_ := file.Close()`.